### DISCUSSION TITLE
Trying to auto-import datetime within neovim using nvim-cmp

### DISCUSSION THREAD

**Discussion by smacz42** [microsoft/pyright#2656] [Q&A] [answered]
Hello all,

I have a setup of neovim where I'm using the language server to provide completions for the auto-fill of nvim-cmp. I'm able to get completions for most what I need, however I was looking into the auto-import features, which I'm trying to understand how it's supposed to work.

So I was thinking about trying out importing datetime. So I started typing `date...` and the only thing that showed up was `DynamicClassAttribute`. I decided to auto-complete it anyways, and it auto-inserted the import line `from types import DynamicClassAttribute`. 

Ok, so I know that it can do auto-import completions, so how can I get it to do datetime?

First, I typed `import date...`, which prompted me to autocomplete `datetime`. And after that, when I was working with datetime objects, I got the correct auto-compltetions.

I guess my question is, why wouldn't it prompt me to autofill (and subsequently auto-import) `datetime` without having to import it first?

**Comment by erictraut** [comment, marked as answer]
Pyright will offer auto-import symbols for modules that have already been imported in other files within your project. So, if you've imported datetime in one file and then go to another file, you will see it offered as an auto-import completion in the second file.

Pylance enhances this capability with a more sophisticated background indexer that offers auto-import for all symbols in your project and all installed modules. This indexing capability is not something we plan to provide in pyright, however. Pyright is focused on type checking functionality. It offers some basic language server features, but the more advanced language server features are provided in pylance.
